packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart

```diff
@@ -0,0 +1,34 @@
+import 'dart:async';
+import 'dart:html' as html show window;
+
+import 'package:connectivity_platform_interface/connectivity_platform_interface.dart';
+import 'package:connectivity_for_web/connectivity_for_web.dart';
+
+/// The web implementation of the ConnectivityPlatform of the Connectivity plugin.
+class DartHtmlConnectivityPlugin extends ConnectivityPlugin {
+  /// Checks the connection status of the device.
+  @override
+  Future<ConnectivityResult> checkConnectivity() async {
+    return html.window.navigator.onLine
```

Comment:
Heads up: these things will become non-null soon. Consider performing null checks using JS-interop, or this hack:

```dart
bool _unsafeIsNull(dynamic object) => object == null;
```

(I'm not sure if this hack will work for long though; dart2js can inline it and decide to elide the null check)

Reply:
Hm `onLine` is a boolean; I hope the whole chain is not-null and defaults to "false" when null safety comes!